kaitai/kaitaistream.cpp

```diff
@@ -344,6 +344,12 @@ uint64_t kaitai::kstream::get_mask_ones(int n) {
 std::string kaitai::kstream::read_bytes(std::streamsize len) {
     std::vector<char> result(len);
 
+    // NOTE: streamsize type is signed, negative values are only *supposed* to not be used.
+    // http://en.cppreference.com/w/cpp/io/streamsize
+    if (len < 0) {
+        throw std::runtime_error("read_bytes: requested a negative amount");
+    }
```

Comment:
`read()` would throw a `std::ios_base::failure` exception on negative amount requested, no need to check here.

Reply:
Okie, do you want me to remove this one then?

Reply:
Don't worry, I'll probably fix all `len`-related checks myself, after the comments requested [here](https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/pull/20#discussion_r180395392).

Reply:
Notice that, even before this check, `len` is used to set the initial the size of the vector...I don't know how it will react to a negative value....

Reply:
Um, we just definitely need more tests for that one, and that question actually still stands for all other target languages.